components/Terminal/index.vue

```diff
@@ -1,150 +1,193 @@
+
+  if (key == "Tab") return;
+  event.preventDefault();
+
+  if (key == "Enter") {
+    const command = activeLineBuffer.value.trim();
+    const pwd = cwd(commandState.value, []);
+    let response;
+    if (command !== "") {
+      response = handleCommand(command);
     }
-    if (key == 'Backspace') {
-        if (inputBuffer.value === '') return
-        activeLineBuffer.value = activeLineBuffer.value.slice(0, -1)
-        inputBuffer.value = activeLineBuffer.value
-        return
+    history.value.push({ input: command, output: response, cwd: pwd });
+
+    inputBuffer.value = "";
+    activeLineBuffer.value = "";
+    historySelectionOffset.value = 0;
+
+    return;
+  }
+  if (key == "Backspace") {
+    if (inputBuffer.value === "") return;
+    activeLineBuffer.value = activeLineBuffer.value.slice(0, -1);
+    inputBuffer.value = activeLineBuffer.value;
+    return;
+  }
+
+  if (key == "ArrowUp") {
+    historySelectionOffset.value++;
+    if (historySelectionOffset.value >= history.value.length) {
+      historySelectionOffset.value = history.value.length;
     }
-
-    if (key == 'ArrowUp') {
-        historySelectionOffset.value++
-        const historySelection = history.value[history.value.length - historySelectionOffset.value]
-        activeLineBuffer.value = historySelection.input
-        return
+    const historySelection = history.value.at(
+      -1 * historySelectionOffset.value
+    );
+    if (!historySelection) {
+      console.error(
+        `something went wrong, debug info: offset: ${historySelectionOffset.value}, history length: ${history.value.length}`
+      );
     }
-    if (key == 'ArrowDown') {
-        historySelectionOffset.value--
-        if (historySelectionOffset.value <= 0) {
-            historySelectionOffset.value = 0
-            activeLineBuffer.value = inputBuffer.value
-        } else {
-            const historySelection = history.value[history.value.length - historySelectionOffset.value]
-            activeLineBuffer.value = historySelection.input
-        }
-        return
+    activeLineBuffer.value = historySelection?.input ?? "";
+    return;
+  }
+
+  if (key == "ArrowDown") {
+    historySelectionOffset.value--;
+
+    if (historySelectionOffset.value <= 0) {
+      historySelectionOffset.value = 0;
+      activeLineBuffer.value = inputBuffer.value;
+    } else {
+      const historySelection = history.value.at(
+        -1 * historySelectionOffset.value
+      );
+      activeLineBuffer.value = historySelection?.input ?? "";
     }
+    return;
+  }
 
-    if (key.length > 1) {
-        console.log(key)
-        return
-    }
+  if (key.length > 1) {
+    console.log(key);
+    return;
+  }
 
-    activeLineBuffer.value += key
-    inputBuffer.value = activeLineBuffer.value
+  activeLineBuffer.value += key;
+  inputBuffer.value = activeLineBuffer.value;
 }
+
+function handleClear(event: KeydownEvent) {
+  event.preventDefault()
+  clearScreen()
+  // HACK
```

Comment:
In `handleInput`, we can detect if `ctrl` is pressed with the key, that might be a less hacky solution?

Reply:
That was my original approach, but I couldn't figure out a way of detecting it without building an entire key press history, which would be overkill

Reply:
[`event.ctrlKey`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/ctrlKey) is `true` if CTRL is pressed, `false` otherwise